lib/astutils.cpp

```diff
@@ -1544,8 +1544,14 @@ bool isUsedAsBool(const Token* const tok, const Settings& settings)
 }
 
 bool compareTokenFlags(const Token* tok1, const Token* tok2, bool macro) {
-    if (macro && (tok1->isExpandedMacro() || tok2->isExpandedMacro() || tok1->isTemplateArg() || tok2->isTemplateArg()))
-        return false;
+    if (macro) {
+        if (tok1->isExpandedMacro() != tok2->isExpandedMacro())
+            return false;
+        if (tok1->isExpandedMacro() && tok1->originalName() != tok2->originalName())
```

Comment:
`if (MACRO == MACRO)` should still result in a warning

Reply:
hmm I am not sure but I don't think originalName checks the macro name? If we check that the macro names in LHS and RHS are the same then I would agree with you.

Reply:
Should probably use `getMacroName()` instead, and add a test. `originalName()` seems to be for typedefs only.

Reply:
Should say "typdefs and other simplifications which aren't macro replacements". We should probably spell that out in a comment for `originalName()`.